src/app/Main/Body/QueryPage/QueryPage-Container.component.tsx

```diff
@@ -207,6 +228,110 @@ export function QueryPageContainerComponent(): JSX.Element {
         setFilter(newFilter);
     };
 
+    const fetchIsolatesAndExportZip = async (): Promise<void> => {

```

Comment:
I recommend a refactoring for the container (it is really big now). One step could be to extract the fetching functions to separate services. E.g. this could look like (simplified): 
```
const fetchIsolatesAndExportZip = async (): Promise<void> => {
  setLoadingIsolates(true);
  await exportIsolatesService(some params);
  setIsLoadingIsolates(false);
}
```

Reply:
New ticket is written.